File: arrays_and_hashing/groupAnagrams.java
import java.util.List;
import java.util.Map;
import java.util.HashMap;
import java.util.ArrayList;
import java.util.Arrays;

class Solution {
  public List<List<String>> groupAnagrams(String[] strs) {
    Map<String, List<String>> con = new HashMap<>();
        
    for (String str : strs) {
      char[] charArray = str.toCharArray();
      Arrays.sort(charArray);
      String strSorted = new String(charArray);
      con.computeIfAbsent(strSorted, k -> new ArrayList<>()).add(str);
    }
    return new ArrayList<>(con.values());
  }
}

File: arrays_and_hashing/groupAnagrams.cpp
#include <vector>
#include <string>
#include <unordered_map>
#include <algorithm>


class Solution {
public:
  vector<vector<string>> groupAnagrams(vector<string>& strs) {
    std::unordered_map<std::string, std::vector<std::string>> con;
    
    for (const std::string& str : strs) {
      std::string strSorted = str;
      std::sort(strSorted.begin(), strSorted.end());
      con[strSorted].emplace_back(str);
    }

    std::vector<std::vector<string>> res;
    for (const auto& entry : con) {
      res.emplace_back(entry.second.begin(), entry.second.end());
    }
    return res;
  }
};